hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/HealthTransformations.scala

```diff
@@ -0,0 +1,114 @@
+package org.broadinstitute.monster.dap
+
+import org.broadinstitute.monster.dogaging.jadeschema.table.HlesHealthCondition
+
+object HealthTransformations {
+
+  /** Parse all health-condition-related fields out of a raw RedCap record. */
+  def mapHealthConditions(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    mapInfectiousDisease(rawRecord)
+
+  /** Generic helper method for creating Hles Health Condition rows. */
+  def createHealthConditionRow(
+    rawRecord: RawRecord,
+    conditionName: String,
+    conditionType: Long,
+    condition: Long,
+    isCongenital: Boolean = false,
+    conditionOtherDescription: Option[String] = None,
+    conditionCause: Option[Long] = None,
+    conditionCauseOtherDescription: Option[String] = None
+  ): Option[HlesHealthCondition] =
+    if (rawRecord.getBoolean(s"hs_${conditionName}")) {
+      Some(
+        HlesHealthCondition(
+          dogId = rawRecord.getRequired("study_id").toLong,
+          hsConditionType = conditionType,
+          hsCondition = condition,
+          hsConditionOtherDescription = conditionOtherDescription,
+          hsConditionIsCongenital = isCongenital,
+          hsConditionCause = conditionCause,
+          hsConditionCauseOtherDescription = conditionCauseOtherDescription,
+          hsDiagnosisYear = rawRecord.getOptionalNumber(s"hs_${conditionName}_year"),
+          hsDiagnosisMonth = rawRecord.getOptionalNumber(s"hs_${conditionName}_month"),
+          hsRequiredSurgeryOrHospitalization =
+            rawRecord.getOptionalNumber(s"hs_${conditionName}_surg"),
+          hsFollowUpOngoing = rawRecord.getOptionalBoolean(s"hs_${conditionName}_fu")
+        )
+      )
+    } else {
+      None
+    }
+
+  /** Parse all infectious disease related fields out of a raw RedCap record. */
+  def mapInfectiousDisease(rawRecord: RawRecord): Iterable[HlesHealthCondition] =
+    if (rawRecord.getBoolean("hs_dx_infectious_yn")) {
+      // iterate over all infectious diseases
+      infectious_diseases.flatMap {
+        // "other" case
+        case ("infect_other", categorical) =>
+          createHealthConditionRow(
+            rawRecord,
+            s"dx_infect_other",
+            1L,
```

Comment:
Worth making this a top-level named constant, since it's used in 2 places?

I'm not sure if this will stay "1" over time, DAP will probably expect the numbering to match the order of questioning in HLES. Having a group of constants will make it easier for us to be sure there are no duplicates & the ordering is what we'd expect.

Reply:
So like a constant `map` of all the top level conditions, referenced where appropriate?

Reply:
I was thinking constants like:
```scala
val EyeCondition = 1L
// moar
val InfectiousDisease = 10L
```

Reply:
The map could also work, but then you end up with repeated magic strings at the lookup points

Reply:
lol we're gonna have so many! But ye, I'm down for just listing constants instead of a map

Reply:
It might be worth making a "constants" object in a separate file to store all the magic numbers & maps. Normally I don't like it, but here it feels potentially justified

Reply:
^I'll leave the constants in here, and if/when someone feels it is getting out of control then we can move it to a separate constants object in another file.